homeassistant/components/airtouch4/climate.py

```diff
@@ -0,0 +1,223 @@
+"""AirTouch 4 component to control of AirTouch 4 Climate Devices."""
+
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_DIFFUSE,
+    FAN_FOCUS,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+AT_TO_HA_STATE = {
+    "Heat": HVAC_MODE_HEAT,
+    "Cool": HVAC_MODE_COOL,
+    "AutoHeat": HVAC_MODE_HEAT_COOL,
+    "AutoCool": HVAC_MODE_HEAT_COOL,
+    "Auto": HVAC_MODE_HEAT_COOL,
```

Comment:
I think I didn't use the auto mode because I thought the developer doc description "The device is set to a schedule, learned behavior, AI." didnt fit the behaviour of the device - setting the device to auto just means that it uses the current temperature and the target temperature to decide which HVAC mode it uses (does that count as AI?) - I agree it probably makes sense to switch to hvac auto - all three of these states (AutoHeat, AutoCool, and Auto) should be the same though - the device reports AutoHeat or AutoCool depending on what decision it makes after being set to Auto (so still the same 'mode')